src/main/java/bio/terra/cli/utils/MountUtils.java

```diff
@@ -0,0 +1,241 @@
+import bio.terra.cli.businessobject.Resource;
+import bio.terra.cli.businessobject.ResourcePropertyNames;
+import bio.terra.cli.businessobject.Workspace;
+import bio.terra.cli.businessobject.mounthandler.GcsFuseMountHandler;
+import bio.terra.cli.businessobject.mounthandler.ResourceMountHandler;
+import bio.terra.cli.businessobject.resource.GcsBucket;
+import bio.terra.cli.businessobject.resource.GcsObject;
+import bio.terra.cli.exception.SystemException;
+import bio.terra.cli.exception.UserActionableException;
+import bio.terra.workspace.model.Folder;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.stream.Stream;
+
+/** This class provides utility methods for mounting and unmount workspace resources */
+public class MountUtils {
+
+  // Directory to mount workspace resources under
+  private static final Path WORKSPACE_DIR = Paths.get(System.getProperty("user.home"), "workspace");
+
+  // Check if the workspace directory exists
+  public static boolean workspaceDirExists() {
+    return Files.exists(WORKSPACE_DIR) && Files.isDirectory(WORKSPACE_DIR);
+  }
+
+  /**
+   * Mounts all mountable resources for a given workspace
+   *
+   * @param ws workspace context
+   */
+  public static void mountResources(Workspace ws, Boolean disableCache) {
+    Map<UUID, Path> resourceMountPaths = getResourceMountPaths();
+
+    // Create root workspace directory if it does not exist
+    try {
+      Files.createDirectories(WORKSPACE_DIR);
+    } catch (IOException e) {
+      throw new SystemException("Error creating workspace directory", e);
+    }
+
+    // Create directories for each resource mount point
+    createResourceDirectories(new ArrayList<>(resourceMountPaths.values()));
+
+    // Mount each resource
+    resourceMountPaths.forEach(
+        (id, mountPath) -> {
+          Resource r = ws.getResource(id);
+          ResourceMountHandler handler = getMountHandler(r, mountPath, disableCache);
+          handler.mount();
+        });
+  }
+
+  /**
+   * Unmount all mountable resources for a given workspace
+   *
+   * @param ws workspace context
+   */
+  public static void unmountResources(Workspace ws) {
+    Map<UUID, Path> resourceMountPaths = getResourceMountPaths();
+
+    // Unmount each resource
+    resourceMountPaths.forEach(
+        (id, mountPath) -> {
+          Resource r = ws.getResource(id);
+          ResourceMountHandler handler = getMountHandler(r, mountPath, null);
+          handler.unmount();
+        });
+
+    // Delete mount point directories if they are empty, so we do not unintentionally delete
+    // bucket files or local files stored at mounted directories
+    deleteResourceDirectories();
```

Comment:
instead of adding the comment, just rename the method to deleteResourceDirectoriesIfEmpty